### PR TITLE
PFMonitor Plugin Individual

### PR DESCRIPTION
**Pull request by pfmonitor**
Upload without the nmap plugin

### CODE REVIEW

**Comment by pfmonitor** [comment]
Obfuscation has also been removed.

File: net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+$cronfile = '/var/cron/tabs/root';
+$crontask = @file_get_contents($cronfile);
+if(strripos($crontask, 'PFMonitor/checkin.php') === false) {
+	echo "PFMonitor Cron job not found!  Installing...\n";
+        system_cron_configure();
+}
+
+// Config file location
+$confdata = "/conf/config.xml";
```
Comment (fabianfrz) [inline net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php]: should be a constant

File: net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+	}
+} else {
+	$data = array('hostname' => $hostname, 'serial' => $serial, 'version' => $version, 'arch' => $arch, 'cpu' => $cpu, 'cores' => $cores, 'uptime' => $uptime, 'config' => $dateconfig, 'states' => $states, 'loads' => $loads, 'memory' => $memory, 'vpnreport' => $vpnreport);
+}
+
+error_reporting(E_NONE);
+$url = 'https://admin.pfmonitor.com/checkin.php';
+$cfgurl = 'https://admin.pfmonitor.com/configupload.php';
+$scrurl = 'https://admin.pfmonitor.com/screenupload.php';
+$tlmurl = 'https://admin.pfmonitor.com/telemetry.php';
```
Comment (fabianfrz) [inline net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php]: looks like constants

File: net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php [hunk 3]
```diff
@@ -0,0 +1,286 @@
+#!/usr/local/bin/php
+<?php
+/*
+ * pfmonitor.checkinopn.php - For OPNSense 18.1.2 and higher
+ *
+ * part of pfMonitor (https://www.pfmonitor.com)
+ * Copyright (c) 2017 MasterX-BKC-
+ * All rights reserved.
```
Comment (fabianfrz) [inline net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php]: license issue

File: net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+	curl_setopt($cfgupload, CURLOPT_POST, true);
+	curl_setopt($cfgupload, CURLOPT_POSTFIELDS, http_build_query($data));
+	curl_setopt($cfgupload, CURLOPT_RETURNTRANSFER, true);
+	curl_setopt($cfgupload, CURLOPT_CONNECTTIMEOUT, 4);
+	curl_setopt($cfgupload, CURLOPT_TIMEOUT, 8);
+	$response = curl_exec($cfgupload);
+	curl_close($cfgupload);
+}
+// Capture Screen
+if(date("i") == "00" || date("i") == "10" || date("i") == "20" || date("i") == "30" || date("i") == "40" || date("i") == "50") {
```
Comment (fabianfrz) [inline net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php]: Switch/case

File: net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+//	curl_setopt($tlmupload, CURLOPT_POSTFIELDS, http_build_query($data));
+//	curl_setopt($tlmupload, CURLOPT_RETURNTRANSFER, true);
+//	curl_setopt($tlmupload, CURLOPT_CONNECTTIMEOUT, 4);
+//	curl_setopt($tlmupload, CURLOPT_TIMEOUT, 8);
+//	$response = curl_exec($tlmupload);
+//	curl_close($tlmupload);
+//}
+
+
+if($command == "success") {
```
Comment (fabianfrz) [inline net-mgmt/pfmonitor/src/opnsense/scripts/PFMonitor/checkin.php]: Switch/case

**Comment by fichtner** [comment]
Closed as discussed in the forum.

**Pull request closed without merging**